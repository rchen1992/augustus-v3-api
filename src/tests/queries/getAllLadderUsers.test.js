const mockLadderRepo = require('@tests/mockRepos/ladder');
const mockLadderUserRepo = require('@tests/mockRepos/ladderUser');
const createLadderService = require('@services/ladderService');
const { testQuery } = require('@tests/utils');

testQuery({
    id: "Get all of a ladder's users",
    query: `
        {
            ladder(id: 1) {
                id
                ladderUsers {
                    id
                    rating
                    joinDate
                    user {
                        id
                        userName
                    }
                }
            }
        }
    `,
    variables: {},
    context: () => {
        return {
            services: {
                ladder: createLadderService(mockLadderRepo, mockLadderUserRepo),
            },
            repos: {
                ladder: mockLadderRepo,
            },
        };
    },
    expected: undefined,
});
