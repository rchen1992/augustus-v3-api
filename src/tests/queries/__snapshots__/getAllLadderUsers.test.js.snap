// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query: Get all of a ladder's users 1`] = `
Object {
  "data": Object {
    "ladder": Object {
      "id": "1",
      "ladderUsers": Array [
        Object {
          "id": "1:1",
          "joinDate": "2018-12-11 09:39:31",
          "rating": 160,
          "user": Object {
            "id": "1",
            "userName": "cblanko0",
          },
        },
        Object {
          "id": "1:2",
          "joinDate": "2018-09-27 20:12:13",
          "rating": 70,
          "user": Object {
            "id": "2",
            "userName": "csaye1",
          },
        },
        Object {
          "id": "1:3",
          "joinDate": "2018-09-27 20:12:13",
          "rating": 110,
          "user": Object {
            "id": "3",
            "userName": "cburchell2",
          },
        },
        Object {
          "id": "1:5",
          "joinDate": "2018-09-27 20:12:13",
          "rating": 140,
          "user": Object {
            "id": "5",
            "userName": "selcott4",
          },
        },
      ],
    },
  },
}
`;
